mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -469,7 +469,32 @@ class Word2VecModel private[spark] (
     this(Word2VecModel.buildWordIndex(model), Word2VecModel.buildWordVectors(model))
   }
 
-  private def cosineSimilarity(v1: Array[Float], v2: Array[Float]): Double = {
+  /**
+   * get cosineSimilarity of two word. assumed to be from the vocabulary and used after model built, otherwise will error out
+   * @param v1 one word from the vocabulary
+   * @param v2 the other word from the vocabulary
+   * @return the cosinesimilarity score in the vector space of the given two words
+   */
+  def cosineSimilarity(v1: String, v2: String): Double = {
+    return cosineSimilarity(getVectors(v1), getVectors(v2))
+  }
+
+  /**
+   * get the built vocabulary from the input
+   * @return a map of word to its index
+   */
+  def getVocabulary:Map[String,Int]={
```

Comment:
Regardless, as others have said, these extra methods should come in a separate PR.